src/irc.py

```diff
@@ -129,52 +126,115 @@ def _install_snap_package(
             logger.exception(error_msg)
             raise InstallError(error_msg) from e
 
-    def _write(self, path: pathlib.Path, source: str) -> None:
-        """Pushes a file to the unit.
+    def configure(
+        self, db: DatasourcePostgreSQL, matrix: DatasourceMatrix, config: CharmConfig
+    ) -> None:
+        """Configure the service.
 
         Args:
-            path: The path of the file
-            source: The contents of the file to be pushed
+            db: the database configuration
+            matrix: the matrix configuration
+            config: the charm configuration
         """
-        path.write_text(source, encoding="utf-8")
-        logger.info("Pushed file %s", path)
-
-    def _generate_PEM_file_local(self) -> str:
-        """Generate the PEM file content.
+        self._generate_pem_file_local()
+        self._generate_app_registration_local(matrix, config)
+        self._eval_conf_local(db, matrix, config)
+
+    def _generate_pem_file_local(self) -> None:
+        """Generate the PEM file content."""
+        pem_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem ]] || "
+            + "openssl genpkey -out {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem "
+            + "-outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:2048",
+        ]
+        logger.info("Creating PEM file for IRC bridge.")
+        subprocess.run(pem_create_command, shell=True, check=True, capture_output=True)  # nosec
```

Comment:
Error handling is missing here

Reply:
What did we say about that one ? I see the comment is closed but without being addressed

Reply:
Error handling is in the next PR, reflected in the unit test as well.